### PR TITLE
docs: gsoc'23 final report - improve development experience 

### PR DESCRIPTION
**Pull request by tanmoysrt**
Added final report of Project - Improve development experience - GSOC'23

### CODE REVIEW

**Review by vedant-jain03 — CHANGES_REQUESTED** [review]
can you please compress the video

**Comment by tanmoysrt** [comment]
> can you please compress the video

Sure !

**Comment by tanmoysrt** [comment]
@vedant-jain03 compressed the videos .

File: content/posts/Tanmoy_Sarkar_Final_Report.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+### Project Description
+---
+CircuitVerse is a Digital Logic Simulator on web. It is an educational tool for students interested in electronics to simulate and make circuits on their browser.
+
+This project focuses on making the process of working and developing with CircuitVerse easier and more efficient. The main goal is to simplify setting up the development environment and working with the code. Additionally, Our aim to enhance the test cases and improve the continuous integration (CI) workflow, which will ultimately reduce the time required for code review by maintainers.
+
+### Project Goals
+---
+Throught this period, we have focused on improving the development experience of CircuitVerse. The main goals of this project are:
```
Comment (vedant-jain03) [inline content/posts/Tanmoy_Sarkar_Final_Report.md:23]:
```suggestion
Throughout this period, we have focused on improving the development experience of CircuitVerse. The main goals of this project are:
```

File: content/posts/Tanmoy_Sarkar_Final_Report.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+##### Pull Requests
+- [Integrate Ruby Debugger](https://github.com/CircuitVerse/CircuitVerse/pull/3760)
+
+
+### Integrate Solargraph LSP & Write YARD Docs
+---
+Solargraph is a Language Server Protocol (LSP) server that enhances autocompletion for Ruby codebases by providing YARD documentation. To further enhance support for Rails codebases, the solargraph-rails gem has been installed and configured. YARD documentation has been generated for the entire codebase to ensure seamless autocompletion.
+
+This integration also supports Docker-based environments, allowing the application to be run in a Docker container while connecting the IDE/Editor's solargraph extension to the Solargraph server via a TCP socket. Initially, there was a significant challenge in this setup. The extension used to send the current directory to the LSP server, but the LSP server had a different path inside the Docker container. However, this issue was successfully resolved by creating a symlink inside the container.
```
Comment (vedant-jain03) [inline content/posts/Tanmoy_Sarkar_Final_Report.md:52]:
```suggestion
This integration also supports Docker-based environment, allowing the application to be run in a Docker container while connecting the IDE/Editor's solargraph extension to the Solargraph server via a TCP socket. Initially, there was a significant challenge in this setup. The extension was used to send the current directory to the LSP server, but the LSP server had a different path inside the Docker container. However, this issue was successfully resolved by creating a symlink inside the container. 
```

**Review by vedant-jain03 — CHANGES_REQUESTED** [review]
just a few comments

File: content/posts/Tanmoy_Sarkar_Final_Report.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+date: 2023-08-27T12:04:13+05:30
+draft: false
+author: Tanmoy Sarkar
+type: post
+---
+
+![gsoc23_final_phase](/images/tanmoy_gsoc23/final_phase_cover_image.png)
+
+
+This blog is dedicated to describe the work we have done and the goals we have accomplished in Google Summer of Code'23.
```
[the remark this answers is not in the document]
Comment (tanmoysrt) [reply content/posts/Tanmoy_Sarkar_Final_Report.md]: updated !

**Review by vedant-jain03 — APPROVED** [review]

**Pull request merged**